packages/core/src/editor/BlockNoteEditor.ts

```diff
@@ -212,6 +216,39 @@ export type BlockNoteEditorOptions<
     string | undefined
   >;
 
+  /**
+   * Custom paste handler that can be used to override the default paste behavior.
+   * @returns The function should return `true` if the paste event was handled, otherwise it should return `false` if it should be canceled or `undefined` if it should be handled by another handler.
+   *
+   * @example
+   * ```ts
+   * pasteHandler: ({ defaultPasteHandler }) => {
+   *   return defaultPasteHandler({ pasteBehavior: "prefer-html" });
+   * }
+   * ```
+   */
+  pasteHandler?: (context: {
+    event: ClipboardEvent;
+    editor: BlockNoteEditor<BSchema, ISchema, SSchema>;
+    /**
+     * The default paste handler
+     * @param context The context object
+     * @returns Whether the paste event was handled or not
+     */
+    defaultPasteHandler: (context?: {
+      /**
+       * Whether to auto-detect markdown in the clipboard and paste that as rich text.
+       * @default true
+       */
+      autoDetectMarkdown?: boolean;
+      /**
+       * Whether to interpret plain text as markdown and paste that as rich text.
+       * @default true
+       */
+      plainTextAsMarkdown?: boolean;
```

Comment:
Imo the difference between `autoDetectMarkdown` and `plainTextAsMarkdown` is not super clear. Don't they kind of do the same thing, except `autoDetectMarkdown` will prioritize `text/plain` as Markdown over `text/html`?

Reply:
I think I would change the options slightly to this:

`plainTextAsMarkdown`: Whether to parse `text/plain` content from the clipboard as Markdown content.

`prioritizeMarkdownOverHTML`: Whether to prioritize Markdown content in `text/plain` over `text/html` when pasting from the clipboard. `plainTextAsMarkdown` must be set to `true` for this to have any effect.

Thoughts on this?